app/celer-g4/GeantStepDiagnostic.hh

```diff
@@ -0,0 +1,73 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2022-2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celer-g4/GeantStepDiagnostic.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+#include <G4Track.hh>
+
+#include "corecel/Types.hh"
+#include "corecel/io/OutputInterface.hh"
+
+namespace celeritas
+{
+namespace app
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Tally the steps per track transported with Geant4 for each particle type.
+ *
+ * For the diagnostic class that collects the same result for tracks
+ * transported with Celeritas, see: \sa StepDiagnostic.
+ */
+class GeantStepDiagnostic final : public OutputInterface
```

Comment:
This class doesn't require anything specific to the celer-g4 app, so let's move it to accel?